### DISCUSSION TITLE
Image SRC Width and Height Issue

### DISCUSSION THREAD

**Discussion by hrodriguez007** [vercel/next.js#43463] [Help] [answered]
Hello, I have been receiving this error when I already set the width and height for the images that I want to upload.
Backstory: I am creating an NFT Marketplace and currently using the test network Localhost:8545 to host it. I will be moving it to Polygon Mumbai but want to test out the create NFT feature first. Might anyone know why this could be happening?

This is the whole error:
Uncaught Error: Image with src "https://vivi-project.infura-ipfs.io/ipfs/QmPHThqtKGAHHgBub542k4bXH9vb55Cc64ZsCyLQYhX3oc" must use "width" and "height" properties or "layout='fill'" property.

This is my image properties:
                         <Image
                            src={fileUrl}
                            alt="Picture of the NFT"
                            className="rounded mt-4"
                            width={350}
                            height={500}
                            //blurDataURL="data:..." automatically provided
                            //placeholder="blur" // Optional blur-up while loading
                          />

**Comment by icyJoseph** [comment, marked as answer]
Are you absolutely sure that this is the image giving the error? What Next version are you using?

Looking at the source code, this case is only hit if you have a form of:  ` <Image src="i.png" />`

**Reply by hrodriguez007** [reply]
I ran the code again and I got the same error. I go through all the steps to create a token and create a market item using Metamask. Just when I think that I have made it, that error pops up and I don't see my created NFT anywhere. I contacted MetaMask and they sent me here instead. 
I also readjusted my height and width to match the image size to see if that made a difference. No luck.

I have Next version 12.3.4. I tried installing Next 13 yesterday but it wasn't too fond of my other dependencies. I was told on Stack Overflow that I might be using the image legacy component but when I try to import it's not there? As in I would need to separately install it? I'm a beginner here and didn't realize that the documentation for image had changed. I am currently using `import Image from 'next/image';`

This is the documentation they referred to me:
[next legacy image docs](https://nextjs.org/docs/api-reference/next/legacy/image)
[next 13 image docs](https://nextjs.org/docs/api-reference/next/image)

**Reply by hrodriguez007** [reply]
Complete error:
`Unhandled Runtime Error
Error: Image with src ["https://vivi-project.infura-ipfs.io/ipfs/Qmc5VkpgsRbiyxG1152WVfnHgZ4caqVLJDSrUPYhX9TdCP"](http://localhost:3000/%22https://vivi-project.infura-ipfs.io/ipfs/Qmc5VkpgsRbiyxG1152WVfnHgZ4caqVLJDSrUPYhX9TdCP%22) must use "width" and "height" properties or "layout='fill'" property.

Call Stack
Image
node_modules/next/dist/client/image.js (304:18)
renderWithHooks
node_modules/react-dom/cjs/react-dom.development.js (16305:0)
mountIndeterminateComponent
node_modules/react-dom/cjs/react-dom.development.js (20074:0)
beginWork
node_modules/react-dom/cjs/react-dom.development.js (21587:0)
HTMLUnknownElement.callCallback
node_modules/react-dom/cjs/react-dom.development.js (4164:0)
Object.invokeGuardedCallbackDev
node_modules/react-dom/cjs/react-dom.development.js (4213:0)
invokeGuardedCallback
node_modules/react-dom/cjs/react-dom.development.js (4277:0)
beginWork$1
node_modules/react-dom/cjs/react-dom.development.js (27451:0)
performUnitOfWork
node_modules/react-dom/cjs/react-dom.development.js (26557:0)
workLoopSync
node_modules/react-dom/cjs/react-dom.development.js (26466:0)
renderRootSync
node_modules/react-dom/cjs/react-dom.development.js (26434:0)
performConcurrentWorkOnRoot
node_modules/react-dom/cjs/react-dom.development.js (25738:0)
workLoop
node_modules/scheduler/cjs/scheduler.development.js (266:0)
flushWork
node_modules/scheduler/cjs/scheduler.development.js (239:0)
MessagePort.performWorkUntilDeadline
node_modules/scheduler/cjs/scheduler.development.js (533:0)`


and 


`The above error occurred in the <Image> component:

    at Image (webpack-internal:///./node_modules/next/dist/client/image.js:27:22)
    at div
    at div
    at div
    at div
    at Home (webpack-internal:///./pages/index.js:44:81)
    at div
    at MyApp (webpack-internal:///./pages/_app.js:18:27)
    at ErrorBoundary (webpack-internal:///./node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:8:20742)
    at ReactDevOverlay (webpack-internal:///./node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:8:23635)
    at Container (webpack-internal:///./node_modules/next/dist/client/index.js:70:9)
    at AppContainer (webpack-internal:///./node_modules/next/dist/client/index.js:216:26)
    at Root (webpack-internal:///./node_modules/next/dist/client/index.js:407:27)

React will try to recreate this component tree from scratch using the error boundary you provided, ErrorBoundary.
window.console.error @ next-dev.js?3515:20
overrideMethod @ react_devtools_backend.js:4026
logCapturedError @ react-dom.development.js?ac89:18687
callback @ react-dom.development.js?ac89:18755
callCallback @ react-dom.development.js?ac89:13923
commitUpdateQueue @ react-dom.development.js?ac89:13944
commitLayoutEffectOnFiber @ react-dom.development.js?ac89:23364
commitLayoutMountEffects_complete @ react-dom.development.js?ac89:24688
commitLayoutEffects_begin @ react-dom.development.js?ac89:24674
commitLayoutEffects @ react-dom.development.js?ac89:24612
commitRootImpl @ react-dom.development.js?ac89:26823
commitRoot @ react-dom.development.js?ac89:26682
finishConcurrentRender @ react-dom.development.js?ac89:25892
performConcurrentWorkOnRoot @ react-dom.development.js?ac89:25809
workLoop @ scheduler.development.js?bcd2:266
flushWork @ scheduler.development.js?bcd2:239
performWorkUntilDeadline @ scheduler.development.js?bcd2:533`

**Reply by icyJoseph** [reply]
I'd say, with this much info that the next/image version doesn't matter. The error you are getting is from the so called legacy next/image, but in the version you have, you don't have to think about this at all.

Could you add me to this repository as a read-only collaborator? Or perhaps provide some screenshots/code snippets?


To re-iterate, the only way to reach that branch of code is:

```ts
  if (layout === 'fill') {
    // <Image src="i.png" layout="fill" />
    // some logic is done here - first branch
  } else if (
    typeof widthInt !== 'undefined' &&
    typeof heightInt !== 'undefined'
  ) {
    // <Image src="i.png" width="100" height="100" />
    // some logic is done here - second branch
  } else {
    // <Image src="i.png" />
    if (process.env.NODE_ENV !== 'production') {
      throw new Error(
        `Image with src "${src}" must use "width" and "height" properties or "layout='fill'" property.`
      )
    }
```

In your snippet, layout is not "fill", so the first branch is not run, and then `width` and `height` are numbers, so they'll be assigned to `widthInt` and `heightInt` and the second branch is not run either, so we end up in the last `else`. Something is not adding up here, and we need more information to move forward.

I'll try with `next@12.3.4` - I observed no issues with:

```jsx
      <Image
        src="https://picsum.photos/200/300"
        alt="A random image"
        width={200}
        height={300}
      />
```

**Reply by hrodriguez007** [reply]
Hello, I have uploaded my repository to GitHub but have it as private for now. I added you to that repository as a collaborator. I would like to preface that I have been referencing another user's code as well. It is a year old so I have had to make some changes of my own. This is the link for the one I am following: [https://github.com/davepartner/opensea-clone] and the youtube video: [https://www.youtube.com/watch?v=7Q5E6RvLlUw&t=1976s]

**Reply by icyJoseph** [reply]
I see this image:

```jsx
    <Image
                  src={nft.image}
                  alt="Picture of the author"
                  // width={500} automatically provided
                  // height={500} automatically provided
                  // blurDataURL="data:..." automatically provided
                  // placeholder="blur" // Optional blur-up while loading
                />
```

Which needs to have `width` at least, found in `page/index.js` do you think that could be the problem?

**Reply by hrodriguez007** [reply]
You are my saving grace! I'm a complete newbie and have been working on the issue for a couple weeks now. Posting on different forums and the like, without much luck and looked at reference documents for Next.js, Infura, Polygon, Tailwind, and Hardhat. You looked at my code and immediately pinpointed what was wrong. It's always the little things that get overlooked. Thanks a million! 👍 
<img width="497" alt="Screen Shot 2022-11-29 at 9 59 33 AM" src="https://user-images.githubusercontent.com/87452210/204580117-071b8806-4c6d-4db6-8602-ebee8f97f2cd.png">

**Reply by icyJoseph** [reply]
Keep up the good spirit! Learning is full of these kind of situations, but every time we learn we progress. Don't forget to revoke me from the repository.

**Comment by exezick** [comment]
You can use the `suppressHydrationWarning`, 
```
<Image
        src="https://picsum.photos/200/300"
        alt="A random image"
        width={200}
        height={300}
        suppressHydrationWarning
      />
```
